lib/engine/game/g_1880/game.rb

```diff
@@ -323,25 +329,70 @@ def init_minors
           game_minors.map { |minor| G1880::Minor.new(**minor) }
         end
 
+        def setup
+          @fully_funded_corporations = []
+          @full_cap_event = false
+          @foreign_investors_can_build = true
+          setup_building_permits
+          setup_unsaleable_shares
+        end
+
         def p1
           company_by_id('P1')
         end
 
         def status_array(corporation)
-          return [] unless corporation.floated?
+          return unless corporation.floated?
+          return if corporation.minor?
 
           ["Building Permits: #{corporation.building_permits}"]
         end
 
+        def non_floated_corporations
+          @corporations.each { |c| yield c if !c.minor? && !c.floated? }
+        end
+
         def corporation_show_individual_reserved_shares?(_corporation)
           false
         end
 
+        def event_all_shares_available!
```

Comment:
Do you think it makes sense to group all the events together?

Reply:
as in one event_<event> method, but still multiple logging entries?

Reply:
multiple event methods, but next to each other in the code.